Open Judge System/Services/OJS.Services/Business/Submissions/ArchivedSubmissions/ArchivedSubmissionsBusinessService.cs

```diff
@@ -1,42 +1,103 @@
 ﻿namespace OJS.Services.Business.Submissions.ArchivedSubmissions
 {
+    using System;
     using System.Linq;
-
     using Hangfire.Server;
-
+    using NPOI.SS.Formula.Functions;
+    using OJS.Common;
+    using OJS.Common.Extensions;
     using OJS.Data.Models;
     using OJS.Services.Common.BackgroundJobs;
+    using OJS.Services.Data.Submissions;
     using OJS.Services.Data.Submissions.ArchivedSubmissions;
 
     public class ArchivedSubmissionsBusinessService : IArchivedSubmissionsBusinessService
     {
         private readonly IArchivedSubmissionsDataService archivedSubmissionsData;
         private readonly ISubmissionsBusinessService submissionsBusiness;
+        private readonly ISubmissionsDataService submissionsData;
         private readonly IHangfireBackgroundJobService backgroundJobs;
 
         public ArchivedSubmissionsBusinessService(
             IArchivedSubmissionsDataService archivedSubmissionsData,
             ISubmissionsBusinessService submissionsBusiness,
+            ISubmissionsDataService submissionsData,
             IHangfireBackgroundJobService backgroundJobs)
         {
             this.archivedSubmissionsData = archivedSubmissionsData;
             this.submissionsBusiness = submissionsBusiness;
+            this.submissionsData = submissionsData;
             this.backgroundJobs = backgroundJobs;
         }
 
-        public void ArchiveOldSubmissions(PerformContext context)
+        public int ArchiveOldSubmissionsDailyBatch(PerformContext context, int limit, int maxSubBatchSize)
+        {
+            var leftover = limit % maxSubBatchSize;
+            var iterations = limit / maxSubBatchSize + (leftover > 0 ? 1 : 0);
```

Comment:
Can this be extracted in a separate calculation method so the ArchiveOldSubmissionsDailyBatch is more easily readable?

Reply:
Extracted new variables and more verbose names to improve the readability of the code.